OpenUpSA/wazimap-ng
# Move log files to S3

## Description
I have added an error handler and helper to upload files to S3. `handle_s3_errors` catches Boto Client errors. `upload_files_to_s3` uploads the error CSV files to S3.

## Related Issue
Fixes #172.

## How to test it locally
1. Enter AWS credentials in your `.env `. 
2. Use production settings (`config/production.py`). Or you might add these lines to `config/local.py`: 

```python
     from configurations import values
    AWS_ACCESS_KEY_ID = Common.get_env_value('AWS_ACCESS_KEY_ID')
    AWS_SECRET_ACCESS_KEY = Common.get_env_value('AWS_SECRET_ACCESS_KEY')

    DEFAULT_FILE_STORAGE = values.Value("django.core.files.storage.FileSystemStorage")
    AWS_STORAGE_BUCKET_NAME = Common.get_env_value('AWS_STORAGE_BUCKET_NAME')
    AWS_S3_REGION_NAME = Common.get_env_value('AWS_S3_REGION_NAME')
    AWS_DEFAULT_ACL = None
```
3. If you added the lines above, change `wazimap_ng/services/general/s3_helpers.py`'s config to `Local`: 
```python
from wazimap_ng.config import Local
from wazimap_ng.general.error_handling import handle_s3_errors


def upload_files_to_s3(files):
    with handle_s3_errors():
        client = boto3.client(
            "s3",
            aws_access_key_id=Local.AWS_ACCESS_KEY_ID,
            aws_secret_access_key=Local.AWS_SECRET_ACCESS_KEY,
            region_name=Local.AWS_S3_REGION_NAME
        )
    for file in files:
        with open(file, "rb") as f:
            file_content = f.read()
            with handle_s3_errors():
                client.put_object(
                    Body=file_content,
                    Key=f"{file}",
                    Bucket=Local.AWS_STORAGE_BUCKET_NAME,
                )
```
3. Try to upload some files with errors in admin page. It is expected to upload the error CSV file to S3.

## Changelog

### Added

### Updated

### Removed


## Checklist


### Pull Request


### Commits


### Code Quality


### Testing
